### PR TITLE
make jetty request header size configurable

### PR DESCRIPTION
**Pull request by emurphy**
Make jetty request header size configurable.

In our case the need arises as our Metabase installation shares a domain with another application that uses AWS Cognito's cookie storage feature, despite its [notoriously large cookies](https://github.com/aws-amplify/amplify-js/issues/1545). A temporal need has also [come up before](https://github.com/metabase/metabase/pull/1820) and will likely come up for other Metabase users as well.

###### Before submitting the PR, please make sure you do the following
~~- [ ] If you're attempting to fix a translation issue, please submit your changes to our [POEditor project](https://poeditor.com/join/project/ynjQmwSsGh) instead of opening a PR.~~
### Tests
~~-  [ ] Run the frontend and integration tests with  `yarn lint && yarn flow && yarn test`)~~
-  [X] If there are changes to the backend codebase, run the backend tests with `lein test && lein lint && ./bin/reflection-linter`

-  [X] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)
(unless it's a tiny documentation change).

### CODE REVIEW

**Comment by flamber** [comment]
This looks like a duplicate of #12082

**Comment by camsaul** [comment]
Supercedes #12802

**Comment by camsaul** [comment]
Closes #9386

File: src/metabase/config.clj [hunk 1]
```diff
@@ -23,6 +23,7 @@
    ;; Jetty Settings. Full list of options is available here: https://github.com/ring-clojure/ring/blob/master/ring-jetty-adapter/src/ring/adapter/jetty.clj
    :mb-jetty-port          "3000"
    :mb-jetty-join          "true"
+   :mb-jetty-request-header-size  "8192"
```
Comment (robdaemon) [inline src/metabase/config.clj:26]: We'd like to not specify the default here and not set the value if it's not overridden, that way if the default changes in the Jetty library in the future, we don't need to update this value. I'll provide an example.

File: src/metabase/server.clj [hunk 2]
```diff
@@ -33,7 +33,8 @@
             :max-threads   (config/config-int :mb-jetty-maxthreads)
             :min-threads   (config/config-int :mb-jetty-minthreads)
             :max-queued    (config/config-int :mb-jetty-maxqueued)
-            :max-idle-time (config/config-int :mb-jetty-maxidletime)})
+            :max-idle-time (config/config-int :mb-jetty-maxidletime)
+            :request-header-size (config/config-int :mb-jetty-request-header-size)})
```
Comment (robdaemon) [inline src/metabase/server.clj:37]: This will require a change to not add to the map if it's not set. Will provide an example.

**Review by robdaemon — CHANGES_REQUESTED** [review]

**Comment by robdaemon** [comment]
Inside `server.clj`, you can do this:

```clj
(defn- jetty-config []
  (cond-> (m/filter-vals
           some?
           {:port          (config/config-int :mb-jetty-port)
            :host          (config/config-str :mb-jetty-host)
            :max-threads   (config/config-int :mb-jetty-maxthreads)
            :min-threads   (config/config-int :mb-jetty-minthreads)
            :max-queued    (config/config-int :mb-jetty-maxqueued)
            :max-idle-time (config/config-int :mb-jetty-maxidletime)})
    (config/config-int :mb-jetty-request-header-size) (assoc :request-header-size (config/config-int
                                                                                   :mb-jetty-request-header-size))
    (config/config-str :mb-jetty-daemon)              (assoc :daemon? (config/config-bool :mb-jetty-daemon))
    (config/config-str :mb-jetty-ssl)                 (-> (assoc :ssl? true)
                                                          (merge (jetty-ssl-config)))))
```

**Review by robdaemon — APPROVED** [review]
Awesome, thanks for the contribution! Once the CI tests finish, I'll merge this.

**Comment by emurphy** [comment]
Thank you! 🎉

**Pull request merged**